README


This archive contains the distribution Dist-Zilla-Plugin-VerifyPhases,
version 0.009:

  Compare data and files at different phases of the distribution build process

This software is copyright (c) 2014 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.025.



Changes
Revision history for Dist-Zilla-Plugin-VerifyPhases

0.009     2014-11-16 04:02:48Z
          - bump required version of Term::ANSIColor for bright colours

0.008     2014-11-15 03:02:07Z
          - fix test failure with older Dist::Zilla and coloured log messages

0.007     2014-11-14 01:34:39Z
          - use colour in log messages for greater visibility

0.006     2014-11-06 04:35:42Z
          - now also checking for premature prereq calculation, given a new
            enough Dist::Zilla

0.005     2014-11-02 05:44:51Z
          - avoid reporting false positives for core zilla attributes that
            were set at construction time (via root config or stash)

0.004     2014-11-01 08:47:41Z
          - fix tests that fail with Dist::Zilla 5.023 (changes to added_by
            semantics)
          - also check the execution order of -NameProvider, -LicenseProvider
            plugins and access time of the abstract, main_module, authors and
            prereqs attributes

0.003     2014-04-06 20:38:06Z
          - also check that -VersionProvider plugins do not run too soon

0.002     2014-03-04 06:37:37Z
          - avoid warning about late modifications to Makefile.PL and
            Build.PL, as this is normal and expected
          - also identify the original filename of a file that has been
            renamed
          - additional checks for files removed outside the pruning phase or
            added after the file gathering phase
          - additional checks for files having their encodings changed outside
            of the file gathering or encoding provider phase
          - at the start of the build, adjust the plugin order so we always
            run last in each phase

0.001     2014-02-23 05:11:23Z
          - Initial release.


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-VerifyPhases was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.016
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.31.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Compare data and files at different phases of the distribution build process",
  "AUTHOR" => "Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-VerifyPhases",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::VerifyPhases",
  "PREREQ_PM" => {
    "Digest::MD5" => 0,
    "Dist::Zilla::Role::AfterBuild" => 0,
    "Dist::Zilla::Role::BeforeBuild" => 0,
    "Dist::Zilla::Role::EncodingProvider" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Dist::Zilla::Role::FilePruner" => 0,
    "List::Util" => "1.33",
    "Moose" => 0,
    "Moose::Util" => 0,
    "Term::ANSIColor" => "3.00",
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Role::FileInjector" => 0,
    "Dist::Zilla::Role::LicenseProvider" => 0,
    "Dist::Zilla::Role::NameProvider" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::VersionProvider" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Path::Tiny" => 0,
    "Software::License::Artistic_2_0" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0,
    "if" => 0
  },
  "VERSION" => "0.009",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Digest::MD5" => 0,
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Role::AfterBuild" => 0,
  "Dist::Zilla::Role::BeforeBuild" => 0,
  "Dist::Zilla::Role::EncodingProvider" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::FileInjector" => 0,
  "Dist::Zilla::Role::FileMunger" => 0,
  "Dist::Zilla::Role::FilePruner" => 0,
  "Dist::Zilla::Role::LicenseProvider" => 0,
  "Dist::Zilla::Role::NameProvider" => 0,
  "Dist::Zilla::Role::PrereqSource" => 0,
  "Dist::Zilla::Role::VersionProvider" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "List::Util" => "1.33",
  "Module::Build::Tiny" => "0.039",
  "Moose" => 0,
  "Moose::Util" => 0,
  "Path::Tiny" => 0,
  "Software::License::Artistic_2_0" => 0,
  "Term::ANSIColor" => "3.00",
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL
# This Build.PL for Dist-Zilla-Plugin-VerifyPhases was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.014
use strict;
use warnings;

my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-VerifyPhases was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.008.
    use strict;
    use warnings;

    use 5.006;
    # use Module::Build::Tiny 0.039;
    Module::Build::Tiny::Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.025.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "Module::Build::Tiny" => "0.039",
        "perl" => "5.006"
      },
      "dist_abstract" => "Compare data and files at different phases of the distribution build process",
      "dist_author" => [
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-VerifyPhases",
      "dist_version" => "0.009",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::VerifyPhases",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Digest::MD5" => 0,
        "Dist::Zilla::Role::AfterBuild" => 0,
        "Dist::Zilla::Role::BeforeBuild" => 0,
        "Dist::Zilla::Role::EncodingProvider" => 0,
        "Dist::Zilla::Role::FileGatherer" => 0,
        "Dist::Zilla::Role::FileMunger" => 0,
        "Dist::Zilla::Role::FilePruner" => 0,
        "List::Util" => "1.33",
        "Moose" => 0,
        "Moose::Util" => 0,
        "Term::ANSIColor" => "3.00",
        "namespace::autoclean" => 0,
        "perl" => "5.006",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "Dist::Zilla::File::InMemory" => 0,
        "Dist::Zilla::Role::FileInjector" => 0,
        "Dist::Zilla::Role::LicenseProvider" => 0,
        "Dist::Zilla::Role::NameProvider" => 0,
        "Dist::Zilla::Role::PrereqSource" => 0,
        "Dist::Zilla::Role::VersionProvider" => 0,
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "Path::Tiny" => 0,
        "Software::License::Artistic_2_0" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::More" => 0,
        "if" => 0,
        "perl" => "5.006"
      }
    );


    my %fallback_build_requires = (
      "Dist::Zilla::File::InMemory" => 0,
      "Dist::Zilla::Role::FileInjector" => 0,
      "Dist::Zilla::Role::LicenseProvider" => 0,
      "Dist::Zilla::Role::NameProvider" => 0,
      "Dist::Zilla::Role::PrereqSource" => 0,
      "Dist::Zilla::Role::VersionProvider" => 0,
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "Path::Tiny" => 0,
      "Software::License::Artistic_2_0" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::More" => 0,
      "if" => 0,
      "perl" => "5.006"
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
